tools/python/src/face_recognition.cpp

```diff
@@ -230,6 +230,37 @@ class face_recognition_model_v1
 
 // ----------------------------------------------------------------------------------------
 
+py::list bottom_up_clustering(py::list descriptors, const int min_num_clusters, const double max_dist)
+{
+    DLIB_CASSERT(min_num_clusters > 0);
+
+    size_t num_descriptors = py::len(descriptors);
+    matrix<float> dist(static_cast<long>(num_descriptors), static_cast<long>(num_descriptors));
```

Comment:
Did you get a compiler warning without adding these `static_casts`?  They shouldn't be needed.

Reply:
Not here, no, but I reused this code from another place where I used this code pattern and must have had a reason to do so there. Happy to remove it here if you'd like though.